doc/releases/migration-guide-3.6.rst

```diff
@@ -56,6 +56,12 @@ Device Drivers and Device Tree
 Power Management
 ================
 
+Shell
+=====
+
+* Subsystem and driver shell modules are now disabled by default. Each required shell module must be
+  explicitly enabled via Kconfig.
```

Comment:
actually, it may help to list what symbols have been disables so one does not have to go through the PR to figure which ones one may want to reenable

Reply:
Added, thanks.